crates/misc/dotnet/src/Engine.cs

```diff
@@ -40,6 +42,36 @@ public Store CreateStore()
             return new Store(this);
         }
 
+        /// <summary>
+        /// Converts the WebAssembly text format to the binary format
+        /// </summary>
+        /// <returns>Returns the binary-encoded wasm module.</returns>
+        public byte[] WatToWasm(string wat)
+        {
+            var watBytes = Encoding.UTF8.GetBytes(wat);
+            var watBytesHandle = GCHandle.Alloc(watBytes, GCHandleType.Pinned);

```

Comment:
In C#, we can actually use `fixed` to pin a pointer to a managed array directly.

There's an example of doing that [here](https://github.com/bytecodealliance/wasmtime/blob/78b32e2527e1d857638f284153c308879828a5cd/crates/misc/dotnet/src/Bindings/FunctionBinding.cs#L250-L257).

Reply:
FYI, it's actually move efficient to use `fixed` than `GCHandle` because the former ends up emitting a special local with a "pinned" attribute that allows the JIT to mark the reference as pinned for the runtime without having to use a `GCHandle`.  But, unlike handles, `fixed` is obviously function-scoped (you don't want to pin objects for very long anyway).

Reply:
Oh I was copying what was happening [here](https://github.com/bytecodealliance/wasmtime/blob/78b32e2527e1d857638f284153c308879828a5cd/crates/misc/dotnet/src/Module.cs#L18), should that also be updated to use `fixed`?

Reply:
Yep, let's fix that too :)  I set a bad example 😞 .

Reply:
hehe np, pushed that up too!